切换输入法程序崩溃

### 摘要 / Summary

切换系统输入法时trime崩溃且无法恢复

### 重现步骤 / Steps to Reproduce

切换系统输入法

### 预期行为 / Expected Behavior

正常来回切换

### 日志 / Log
```
--------- Device Info
OS Name: PJF110_15.0.0.701(CN01)
OS Version: 6.1.75-android14-11-o-g74387f2552b6 (U.1cf6eb4_1-8c69)
OS API Level: 35
Device: OP5CFBL1
Model (product): PJF110 (PJF110)
Manufacturer: OnePlus
Tags: release-keys
Screen Size: 1080 x 2245
Screen Density: 3.0
Screen orientation: Portrait
--------- Build Info
Package Name: com.osfans.trime
Builder: (Unknown)
Version Code: 20250301
Version Name: v3.3.3-0-g9e3689b3-release
Build Time: 2025-03-03T02:08:54Z
Build Git Hash: 9e3689b34c8fcf27bfa3f1e11cb3d79a33ec11c3
--------- Crash stacktrace
androidx.startup.StartupException: lateinit property _activeTheme has not been initialized
	at com.osfans.trime.data.theme.ThemeManager.getActiveTheme(SourceFile:8)
	at com.osfans.trime.ime.core.TrimeInputMethodService.onCreateInputView(SourceFile:11)
	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2449)
	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:3307)
	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:3224)
	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:1051)
	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:1012)
	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:282)
	at android.os.Looper.loop(Looper.java:387)
	at android.app.ActivityThread.main(ActivityThread.java:9500)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)

--------- beginning of main
04-05 06:20:56.163 I/om.osfans.trime(11909): isSwordfishLaunchAppEnabled_ 1 isSwordfishInstallAppEnabled_ 1
04-05 06:20:56.164 E/om.osfans.trime(11909): Unknown bits set in runtime_flags: 0x40000000
04-05 06:20:56.164 I/om.osfans.trime(11909): Using CollectorTypeCMC GC.
04-05 06:20:56.168 E/om.osfans.trime(11909): Not starting debugger since process cannot load the jdwp agent.
04-05 06:20:56.171 D/nativeloader(11909): Load libframework-connectivity-tiramisu-jni.so using APEX ns com_android_tethering for caller /apex/com.android.tethering/javalib/framework-connectivity-t.jar: ok
04-05 06:20:56.173 W/libc    (11909): Access denied finding property "persist.vendor.sys.activitylog"
04-05 06:20:56.173 I/OneTrace(11909): Mark active for pid=11909? true
04-05 06:20:56.173 D/cutils-dev(11909): otrace_set_tracing_enabled? true tag:0
04-05 06:20:56.174 D/cutils-dev(11909): properties changed in otrace_seq_number_changed!
04-05 06:20:56.180 D/OplusActivityManager(11909): get AMS extension: android.os.BinderProxy@5a62422
04-05 06:20:56.183 D/CompactWindowAppManager(11909): initCompactApplicationInfo CompactMode: , NormalMode: 
04-05 06:20:56.185 D/OplusScrollToTopManager(11909): setIsInWhiteList false
04-05 06:20:56.186 D/OplusViewDebugManager(11909): OplusViewDebugManager Constructor android.view.debug.OplusViewDebugManager@8415c70 [mHasViewDebugProperty true] [mHasMsgDebugProperty false] [DEBUG false] [EXTRAINFOENABLE false] [LIGHT_OS false]
04-05 06:20:56.188 D/CompatChangeReporter(11909): Compat change id reported: 242716250; UID 10410; state: ENABLED
04-05 06:20:56.190 W/libc    (11909): Access denied finding property "ro.odm.prev.product.name"
04-05 06:20:56.192 D/ApplicationLoaders(11909): Returning zygote-cached class loader: /system_ext/framework/androidx.window.extensions.jar
04-05 06:20:56.192 D/ApplicationLoaders(11909): Returning zygote-cached class loader: /system_ext/framework/androidx.window.sidecar.jar
04-05 06:20:56.197 D/nativeloader(11909): Configuring clns-7 for other apk /data/app/~~25BxDPyfS87hcutOFNumVA==/com.osfans.trime-h5jny3xUcnEuhRzqM_AKfw==/base.apk. target_sdk_version=34, uses_libraries=, library_path=/data/app/~~25BxDPyfS87hcutOFNumVA==/com.osfans.trime-h5jny3xUcnEuhRzqM_AKfw==/lib/arm64:/data/app/~~25BxDPyfS87hcutOFNumVA==/com.osfans.trime-h5jny3xUcnEuhRzqM_AKfw==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.osfans.trime
04-05 06:20:56.204 V/GraphicsEnvironment(11909): Currently set values for:
04-05 06:20:56.205 V/GraphicsEnvironment(11909):   angle_gl_driver_selection_pkgs=[]
04-05 06:20:56.205 V/GraphicsEnvironment(11909):   angle_gl_driver_selection_values=[]
04-05 06:20:56.205 V/GraphicsEnvironment(11909): Global.Settings values are invalid: number of packages: 0, number of values: 0
04-05 06:20:56.205 V/GraphicsEnvironment(11909): Updatable production driver is not supported on the device.
--------- beginning of system
04-05 06:20:56.206 D/ResourcesManagerExtImpl(11909): applyConfigurationToAppResourcesLocked app.getDisplayId() return callback.displayId:-1
04-05 06:20:56.209 D/WM-WrkMgrInitializer(11909): Initializing WorkManager with default configuration.
04-05 06:20:56.210 D/WM-PackageManagerHelper(11909): Skipping component enablement for androidx.work.impl.background.systemjob.SystemJobService
04-05 06:20:56.210 D/WM-Schedulers(11909): Created SystemJobScheduler and enabled SystemJobService
04-05 06:20:56.224 D/OplusInputMethodUtil(11909): init sDebug to false, init sDebugIme to false, init sAlwaysOn to false
04-05 06:20:56.224 D/OplusInputMethodUtil(11909): updateDebugToClass InputMethodService.DEBUG = false
04-05 06:20:56.232 D/nativeloader(11909): Load /data/app/~~25BxDPyfS87hcutOFNumVA==/com.osfans.trime-h5jny3xUcnEuhRzqM_AKfw==/lib/arm64/librime_jni.so using ns clns-7 from class loader (caller=/data/app/~~25BxDPyfS87hcutOFNumVA==/com.osfans.trime-h5jny3xUcnEuhRzqM_AKfw==/base.apk): ok
04-05 06:20:56.236 D/OplusInputMethodUtil(11909): updateDebugToClass InputMethodManager.DEBUG = false
04-05 06:20:56.236 D/OplusInputMethodUtil(11909): updateDebugToClass ImeFocusController.DEBUG = false
04-05 06:20:56.236 D/OplusInputMethodUtil(11909): updateDebugToClass InsetsController.DEBUG = false
04-05 06:20:56.237 D/OplusInputMethodUtil(11909): updateDebugToClass BaseInputConnection.DEBUG = false
04-05 06:20:56.240 I/rime.trime(11909): registering core components.
04-05 06:20:56.240 I/rime.trime(11909): registering component: config_builder
04-05 06:20:56.240 I/rime.trime(11909): registering component: config
04-05 06:20:56.240 I/rime.trime(11909): registering component: schema
04-05 06:20:56.240 I/rime.trime(11909): registering component: user_config
04-05 06:20:56.240 I/rime.trime(11909): registering components from module 'dict'.
04-05 06:20:56.240 I/rime.trime(11909): registering component: tabledb
04-05 06:20:56.240 I/rime.trime(11909): registering component: stabledb
04-05 06:20:56.240 I/rime.trime(11909): registering component: plain_userdb
04-05 06:20:56.240 I/rime.trime(11909): registering component: userdb
04-05 06:20:56.240 I/rime.trime(11909): registering component: corrector
04-05 06:20:56.240 I/rime.trime(11909): registering component: dictionary
04-05 06:20:56.240 I/rime.trime(11909): registering component: reverse_lookup_dictionary
04-05 06:20:56.240 I/rime.trime(11909): registering component: user_dictionary
04-05 06:20:56.240 I/rime.trime(11909): registering component: userdb_recovery_task
04-05 06:20:56.240 I/rime.trime(11909): registering components from module 'gears'.
04-05 06:20:56.240 I/rime.trime(11909): registering component: ascii_composer
04-05 06:20:56.240 I/rime.trime(11909): registering component: chord_composer
04-05 06:20:56.240 I/rime.trime(11909): registering component: express_editor
04-05 06:20:56.240 I/rime.trime(11909): registering component: fluid_editor
04-05 06:20:56.240 I/rime.trime(11909): registering component: fluency_editor
04-05 06:20:56.240 I/rime.trime(11909): registering component: key_binder
04-05 06:20:56.240 I/rime.trime(11909): registering component: navigator
04-05 06:20:56.240 I/rime.trime(11909): registering component: punctuator
04-05 06:20:56.240 I/rime.trime(11909): registering component: recognizer
04-05 06:20:56.240 I/rime.trime(11909): registering component: selector
04-05 06:20:56.240 I/rime.trime(11909): registering component: speller
04-05 06:20:56.240 I/rime.trime(11909): registering component: shape_processor
04-05 06:20:56.240 I/rime.trime(11909): registering component: abc_segmentor
04-05 06:20:56.240 I/rime.trime(11909): registering component: affix_segmentor
04-05 06:20:56.240 I/rime.trime(11909): registering component: ascii_segmentor
04-05 06:20:56.240 I/rime.trime(11909): registering component: matcher
04-05 06:20:56.240 I/rime.trime(11909): registering component: punct_segmentor
04-05 06:20:56.240 I/rime.trime(11909): registering component: fallback_segmentor
04-05 06:20:56.240 I/rime.trime(11909): registering component: echo_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: punct_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: table_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: script_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: r10n_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: reverse_lookup_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: schema_list_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: switch_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: history_translator
04-05 06:20:56.240 I/rime.trime(11909): registering component: simplifier
04-05 06:20:56.240 I/rime.trime(11909): registering component: uniquifier
04-05 06:20:56.240 I/rime.trime(11909): registering component: charset_filter
04-05 06:20:56.240 I/rime.trime(11909): registering component: cjk_minifier
04-05 06:20:56.240 I/rime.trime(11909): registering component: reverse_lookup_filter
04-05 06:20:56.240 I/rime.trime(11909): registering component: single_char_filter
04-05 06:20:56.240 I/rime.trime(11909): registering component: shape_formatter
04-05 06:20:56.240 I/rime.trime(11909): registering components from module 'lua'.
04-05 06:20:56.241 I/oplus.android.OplusFrameworkFactoryImpl(11909): Unknow feature:IOplusTextViewRTLUtilForUG
04-05 06:20:56.242 D/nativeloader(11909): Load liboplushwui_jni.so using system ns (caller=/system/framework/oplus-framework.jar): ok
04-05 06:20:56.255 E/rime.trime(11909): unknown deployment task: config_file_update
04-05 06:20:56.255 E/[main]  (11909): java.lang.IllegalArgumentException: Failed to deploy theme config file 'tongwenfeng.trime'
04-05 06:20:56.255 E/[main]  (11909): 	at com.osfans.trime.data.theme.Theme.<init>(SourceFile:1104)
04-05 06:20:56.255 E/[main]  (11909): 	at com.osfans.trime.data.theme.ThemeManager.init(SourceFile:13)
04-05 06:20:56.255 E/[main]  (11909): 	at com.osfans.trime.ime.core.TrimeInputMethodService.onCreate(SourceFile:168)
04-05 06:20:56.255 E/[main]  (11909): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5329)
04-05 06:20:56.255 E/[main]  (11909): 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
04-05 06:20:56.255 E/[main]  (11909): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2682)
04-05 06:20:56.255 E/[main]  (11909): 	at android.os.Handler.dispatchMessage(Handler.java:107)
04-05 06:20:56.255 E/[main]  (11909): 	at android.os.Looper.loopOnce(Looper.java:282)
04-05 06:20:56.255 E/[main]  (11909): 	at android.os.Looper.loop(Looper.java:387)
04-05 06:20:56.255 E/[main]  (11909): 	at android.app.ActivityThread.main(ActivityThread.java:9500)
04-05 06:20:56.255 E/[main]  (11909): 	at java.lang.reflect.Method.invoke(Native Method)
04-05 06:20:56.255 E/[main]  (11909): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
04-05 06:20:56.255 E/[main]  (11909): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)
04-05 06:20:56.256 I/[main]  (11909): BackgroundSyncWork scheduled!
04-05 06:20:56.257 I/[main]  (11909): BackgroundSyncWork canceled!
04-05 06:20:56.273 D/nativeloader(11909): Load libSchedAssistJni.so using system ns (caller=/system/framework/oplus-framework.jar): ok
04-05 06:20:56.273 D/ExtensionsLoader(11909): createInstance(64bit) : createExtendedFactory
04-05 06:20:56.274 D/ExtensionsLoader(11909): Opened libSchedAssistExtImpl.so
04-05 06:20:56.274 E/SchedAssist(11909): sysOpen failed with error=Permission denied
04-05 06:20:56.274 E/SchedAssist(11909): open sharedFd failed with error=Permission denied
04-05 06:20:56.274 E/SchedAssist(11909): open sharedFd failed with error=Permission denied
04-05 06:20:56.274 E/SchedAssist(11909): open sharedFd failed with error=Permission denied
04-05 06:20:56.274 E/SchedAssist(11909): open sharedFd failed with error=Permission denied
04-05 06:20:56.276 D/OplusGraphicsEvent(11909): OplusGraphicsEventListener create now!
04-05 06:20:56.277 D/OplusGraphicsEvent(11909): addEventListener success!
04-05 06:20:56.279 D/ScrollOptim [ConfigManager](11909): initOptConfig: pkg= com.osfans.trime, mOptEnable= true, mAnimAheadEnable= true, mInsertDefaultNum= 1, mFrameInsertEnableList= {}, mSCEnableList= []
04-05 06:20:56.281 D/WindowManager(11909): Add to mViews: com.android.internal.policy.DecorView{c6c8c45 I.E...... R.....I. 0,0-0,0 alpha=1.0 viewInfo = }[InputMethod],pkg= null
04-05 06:20:56.282 I/HWUI    (11909): SupportApps size 0
04-05 06:20:56.283 I/rime.trime(11909): registering component: lua_translator
04-05 06:20:56.283 I/rime.trime(11909): registering component: lua_filter
04-05 06:20:56.283 I/rime.trime(11909): registering component: lua_segmentor
04-05 06:20:56.283 I/rime.trime(11909): registering component: lua_processor
04-05 06:20:56.283 I/rime.trime(11909): registering components from module 'grammar'.
04-05 06:20:56.283 I/rime.trime(11909): registering component: grammar
04-05 06:20:56.283 I/rime.trime(11909): registering component: predictor
04-05 06:20:56.283 I/rime.trime(11909): registering component: predict_translator
04-05 06:20:56.283 I/rime.trime(11909): registering components from module 'levers'.
04-05 06:20:56.283 I/rime.trime(11909): registering component: detect_modifications
04-05 06:20:56.283 I/rime.trime(11909): registering component: installation_update
04-05 06:20:56.283 I/rime.trime(11909): registering component: workspace_update
04-05 06:20:56.283 I/rime.trime(11909): registering component: schema_update
04-05 06:20:56.283 I/rime.trime(11909): registering component: config_file_update
04-05 06:20:56.283 I/rime.trime(11909): registering component: prebuild_all_schemas
04-05 06:20:56.283 I/rime.trime(11909): registering component: user_dict_upgrade
04-05 06:20:56.283 I/rime.trime(11909): registering component: cleanup_trash
04-05 06:20:56.283 I/rime.trime(11909): registering component: user_dict_sync
04-05 06:20:56.283 I/rime.trime(11909): registering component: backup_config_files
04-05 06:20:56.283 I/rime.trime(11909): registering component: clean_old_log_files
04-05 06:20:56.283 I/rime.trime(11909): updating rime installation info.
04-05 06:20:56.284 I/rime.trime(11909): loading config file '/storage/emulated/0/rime/installation.yaml'.
04-05 06:20:56.284 D/ExtensionsLoader(11909): createInstance(64bit) : createExtendedFactory
04-05 06:20:56.284 D/ExtensionsLoader(11909): Opened libSchedAssistExtImpl.so
04-05 06:20:56.285 I/rime.trime(11909): installation info exists. installation id: oneplus
04-05 06:20:56.285 I/rime.trime(11909): sync dir: /storage/emulated/0/rime/sync
04-05 06:20:56.285 I/rime.trime(11909): previous distribution: trime
04-05 06:20:56.285 I/rime.trime(11909): previous distribution version: v3.3.3-0-g9e3689b3
04-05 06:20:56.285 I/rime.trime(11909): previous Rime version: 1.13.1
04-05 06:20:56.285 D/ExtensionsLoader(11909): createInstance(64bit) : createExtendedFactory
04-05 06:20:56.286 D/ExtensionsLoader(11909): Opened libhwuiextimpl.so
04-05 06:20:56.286 D/skia    (11909): setRusSupported enable 0
04-05 06:20:56.286 D/ExtensionsLoader(11909): createInstance(64bit) : createExtendedFactory
04-05 06:20:56.286 D/ExtensionsLoader(11909): Opened libSchedAssistExtImpl.so
04-05 06:20:56.292 I/rime.trime(11909): loading config file '/storage/emulated/0/rime/user.yaml'.
04-05 06:20:56.293 I/rime.trime(11909): loading config file '/storage/emulated/0/rime/build/default.yaml'.
04-05 06:20:56.295 D/ViewRootImplExtImpl(11909): onDisplayChanged -1 for VRI android.view.ViewRootImpl@2cea7c1
04-05 06:20:56.296 I/rime.trime(11909): starting engine.
04-05 06:20:56.296 I/rime.trime(11909): loading config file '/storage/emulated/0/rime/user.yaml'.
04-05 06:20:56.297 I/rime.trime(11909): updated option: ascii_punct
04-05 06:20:56.297 I/rime.trime(11909): updated option: emoji
04-05 06:20:56.297 I/rime.trime(11909): updated option: full_shape
04-05 06:20:56.297 I/rime.trime(11909): updated option: search_single_char
04-05 06:20:56.297 I/rime.trime(11909): updated option: traditionalization
04-05 06:20:56.297 I/rime.trime(11909): loading config file '/storage/emulated/0/rime/build/t9.schema.yaml'.
04-05 06:20:56.299 E/OplusBracketLog(11909): [OplusViewMirrorManager] updateHostViewRootIfNeeded, not support android.view.ViewRootImpl@2cea7c1
04-05 06:20:56.310 I/DynamicFramerate [FRTCConfigManager](11909): initFrameRateConfig: 
04-05 06:20:56.310 I/DynamicFramerate [FRTCConfigManager](11909): 	levels: [120, 60]
04-05 06:20:56.310 I/DynamicFramerate [FRTCConfigManager](11909): 	thresholds: [200]
04-05 06:20:56.311 I/DynamicFramerate [FRTCConfigManager](11909): FRTCConfigManager: FRTC_CAPABILITY = 120, package name = com.osfans.trime, WINDOW_ANIMATION_SPEED_RATE = 10, SCROLL_ANIMATION_SPEED_RATE = 20, PACKAGE_ENABLE = false
04-05 06:20:56.311 I/DynamicFramerate [DynamicFrameRateController](11909): init info: mPackageName = com.osfans.trime, mIsEnabled = false
04-05 06:20:56.316 I/rime.trime(11909): updated option: _auto_commit
04-05 06:20:56.316 I/rime.trime(11909): loading dictionary 'rime_ice'.
04-05 06:20:56.316 I/rime.trime(11909): loading table file: /storage/emulated/0/rime/build/rime_ice.table.bin
04-05 06:20:56.322 I/rime.trime(11909): loading prism file: /storage/emulated/0/rime/build/t9.prism.bin
04-05 06:20:56.323 I/rime.trime(11909): found double array image of size 6656.
04-05 06:20:56.465 I/rime.trime(11909): stabledb 'custom_phrase_t9' does not exist.
04-05 06:20:56.465 I/rime.trime(11909): initializing opencc: /storage/emulated/0/rime/opencc/emoji.json
04-05 06:20:56.513 I/rime.trime(11909): initializing opencc: /storage/emulated/0/Android/data/com.osfans.trime/files/shared/opencc/s2t.json
04-05 06:20:56.566 I/rime.trime(11909): ConcreteEngine::InitializeOptions
04-05 06:20:56.566 I/rime.trime(11909): found switch option: ascii_mode, reset: -1
04-05 06:20:56.566 I/rime.trime(11909): found switch option: ascii_punct, reset: -1
04-05 06:20:56.566 I/rime.trime(11909): found switch option: traditionalization, reset: -1
04-05 06:20:56.566 I/rime.trime(11909): found switch option: emoji, reset: 1
04-05 06:20:56.566 I/rime.trime(11909): updated option: emoji
04-05 06:20:56.566 I/rime.trime(11909): found switch option: full_shape, reset: -1
04-05 06:20:56.566 I/rime.trime(11909): found switch option: search_single_char, reset: -1
04-05 06:20:56.568 I/[rime-main](11909): Loading config file /storage/emulated/0/rime/build/default.yaml
04-05 06:20:56.573 I/[rime-main](11909): Loading config file /storage/emulated/0/rime/build/t9.schema.yaml
04-05 06:20:58.332 D/HWUI    (11909): RenderProxy::destroy: this=0xb400007b5ffa1900, mContext=0xb400007a00510000
04-05 06:20:58.332 D/HWUI    (11909): SkiaOpenGLPipeline::setSurface: this=0xb400007a004d1a00, surface=NULL
04-05 06:21:05.856 D/AndroidRuntime(11909): Shutting down VM
--------- beginning of crash
04-05 06:21:05.856 E/AndroidRuntime(11909): FATAL EXCEPTION: main
04-05 06:21:05.856 E/AndroidRuntime(11909): Process: com.osfans.trime, PID: 11909
04-05 06:21:05.856 E/AndroidRuntime(11909): androidx.startup.StartupException: lateinit property _activeTheme has not been initialized
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at com.osfans.trime.data.theme.ThemeManager.getActiveTheme(SourceFile:8)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at com.osfans.trime.ime.core.TrimeInputMethodService.onCreateInputView(SourceFile:11)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2449)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:3307)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:3224)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:1051)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:1012)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.os.Handler.dispatchMessage(Handler.java:107)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.os.Looper.loopOnce(Looper.java:282)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.os.Looper.loop(Looper.java:387)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at android.app.ActivityThread.main(ActivityThread.java:9500)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at java.lang.reflect.Method.invoke(Native Method)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
04-05 06:21:05.856 E/AndroidRuntime(11909): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)
04-05 06:21:05.879 I/om.osfans.trime(11909): System.exit called, status: 10
04-05 06:21:05.879 I/AndroidRuntime(11909): VM exiting with result code 10, cleanup skipped.
```

### 截图 / Screenshot

_No response_

### 附加信息 / Additional Context

_No response_

### 系统版本 / OS Version

android 15

### 应用版本 / App Version

v3.3.3-0-g9e3689b3-release

### 构建信息 / Build Information

构建者：(Unknown)
Git 仓库：https://github.com/osfans/trime
构建 Git 哈希：9e3689b34c8fcf27bfa3f1e11cb3d79a33ec11c3
构建时间：2025年3月3日 10:08:54

更多日志如下
```
--------- Device Info
OS Name: PJF110_15.0.0.701(CN01)
OS Version: 6.1.75-android14-11-o-g74387f2552b6 (U.1cf6eb4_1-8c69)
OS API Level: 35
Device: OP5CFBL1
Model (product): PJF110 (PJF110)
Manufacturer: OnePlus
Tags: release-keys
Screen Size: 1080 x 2245
Screen Density: 3.0
Screen orientation: Portrait
--------- Build Info
Package Name: com.osfans.trime
Builder: (Unknown)
Version Code: 20250301
Version Name: v3.3.3-0-g9e3689b3-release
Build Time: 2025-03-03T02:08:54Z
Build Git Hash: 9e3689b34c8fcf27bfa3f1e11cb3d79a33ec11c3
--------- Crash stacktrace
androidx.startup.StartupException: lateinit property _activeTheme has not been initialized
	at com.osfans.trime.data.theme.ThemeManager.getActiveTheme(SourceFile:8)
	at com.osfans.trime.ime.core.TrimeInputMethodService.onCreateInputView(SourceFile:11)
	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2449)
	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:3307)
	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:3224)
	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:1051)
	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:1012)
	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:282)
	at android.os.Looper.loop(Looper.java:387)
	at android.app.ActivityThread.main(ActivityThread.java:9500)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)

--------- beginning of main
04-05 06:24:02.750 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/rime_ice.schema.yaml'.
04-05 06:24:02.768 I/rime.trime(17538): preparing dictionary 'rime_ice'.
04-05 06:24:02.768 I/rime.trime(17538): compiling dictionary for /storage/emulated/0/rime/build/rime_ice.schema.yaml
04-05 06:24:03.211 I/rime.trime(17538): loading table file: /storage/emulated/0/rime/build/rime_ice.table.bin
04-05 06:24:03.215 I/rime.trime(17538): loading prism file: /storage/emulated/0/rime/build/rime_ice.prism.bin
04-05 06:24:03.219 I/rime.trime(17538): found double array image of size 2304.
04-05 06:24:03.219 I/rime.trime(17538): /storage/emulated/0/rime/rime_ice.dict.yaml[6 file(s)] (2995241493)
04-05 06:24:03.219 I/rime.trime(17538): /storage/emulated/0/rime/build/rime_ice.schema.yaml (22040509)
04-05 06:24:03.219 I/rime.trime(17538): loading reversedb: /storage/emulated/0/rime/build/rime_ice.reverse.bin
04-05 06:24:03.221 I/rime.trime(17538): dictionary 'rime_ice' is ready.
04-05 06:24:03.222 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/rime_ice.schema.yaml'.
04-05 06:24:03.240 I/rime.trime(17538): schema: melt_eng
04-05 06:24:03.241 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/melt_eng.schema.yaml'.
04-05 06:24:03.245 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/melt_eng.schema.yaml'.
04-05 06:24:03.253 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/melt_eng.schema.yaml'.
04-05 06:24:03.258 I/rime.trime(17538): preparing dictionary 'melt_eng'.
04-05 06:24:03.258 I/rime.trime(17538): compiling dictionary for /storage/emulated/0/rime/build/melt_eng.schema.yaml
04-05 06:24:03.269 I/rime.trime(17538): loading table file: /storage/emulated/0/rime/build/melt_eng.table.bin
04-05 06:24:03.272 I/rime.trime(17538): loading prism file: /storage/emulated/0/rime/build/melt_eng.prism.bin
04-05 06:24:03.274 I/rime.trime(17538): found double array image of size 672000.
04-05 06:24:03.275 I/rime.trime(17538): /storage/emulated/0/rime/melt_eng.dict.yaml[3 file(s)] (1308037836)
04-05 06:24:03.275 I/rime.trime(17538): /storage/emulated/0/rime/build/melt_eng.schema.yaml (1961309829)
04-05 06:24:03.275 I/rime.trime(17538): loading reversedb: /storage/emulated/0/rime/build/melt_eng.reverse.bin
04-05 06:24:03.277 I/rime.trime(17538): dictionary 'melt_eng' is ready.
04-05 06:24:03.277 I/rime.trime(17538): schema: radical_pinyin
04-05 06:24:03.278 W/rime.trime(17538): missing input schema; skipped unsatisfied dependency: radical_pinyin
04-05 06:24:03.278 I/rime.trime(17538): schema: t9
04-05 06:24:03.278 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/t9.schema.yaml'.
04-05 06:24:03.283 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/t9.schema.yaml'.
04-05 06:24:03.304 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/t9.schema.yaml'.
04-05 06:24:03.323 I/rime.trime(17538): preparing dictionary 'rime_ice'.
04-05 06:24:03.323 I/rime.trime(17538): compiling dictionary for /storage/emulated/0/rime/build/t9.schema.yaml
04-05 06:24:03.672 I/rime.trime(17538): loading table file: /storage/emulated/0/rime/build/rime_ice.table.bin
04-05 06:24:03.678 I/rime.trime(17538): loading prism file: /storage/emulated/0/rime/build/t9.prism.bin
04-05 06:24:03.678 I/rime.trime(17538): found double array image of size 6656.
04-05 06:24:03.678 I/rime.trime(17538): /storage/emulated/0/rime/rime_ice.dict.yaml[6 file(s)] (2995241493)
04-05 06:24:03.678 I/rime.trime(17538): /storage/emulated/0/rime/build/t9.schema.yaml (459461029)
04-05 06:24:03.679 I/rime.trime(17538): loading reversedb: /storage/emulated/0/rime/build/rime_ice.reverse.bin
04-05 06:24:03.682 I/rime.trime(17538): dictionary 'rime_ice' is ready.
04-05 06:24:03.683 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/t9.schema.yaml'.
04-05 06:24:03.702 I/rime.trime(17538): finished updating schemas: 3 success, 0 failure.
04-05 06:24:03.702 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/user.yaml'.
04-05 06:24:03.705 I/rime.trime(17538): write: var/last_build_time
04-05 06:24:03.705 I/rime.trime(17538): saving config file '/storage/emulated/0/rime/user.yaml'.
04-05 06:24:03.712 I/rime.trime(17538): clean up trash.
04-05 06:24:03.714 I/rime.trime(17538): 3 tasks ran: 3 success, 0 failure.
04-05 06:24:03.715 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/default.yaml'.
04-05 06:24:03.718 I/rime.trime(17538): starting engine.
04-05 06:24:03.718 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/user.yaml'.
04-05 06:24:03.719 I/rime.trime(17538): updated option: ascii_punct
04-05 06:24:03.719 I/rime.trime(17538): updated option: emoji
04-05 06:24:03.719 I/rime.trime(17538): updated option: full_shape
04-05 06:24:03.719 I/rime.trime(17538): updated option: search_single_char
04-05 06:24:03.719 I/rime.trime(17538): updated option: traditionalization
04-05 06:24:03.719 I/rime.trime(17538): loading config file '/storage/emulated/0/rime/build/t9.schema.yaml'.
04-05 06:24:03.740 I/rime.trime(17538): updated option: _auto_commit
04-05 06:24:03.740 I/rime.trime(17538): loading dictionary 'rime_ice'.
04-05 06:24:03.740 I/rime.trime(17538): loading table file: /storage/emulated/0/rime/build/rime_ice.table.bin
04-05 06:24:03.744 I/rime.trime(17538): loading prism file: /storage/emulated/0/rime/build/t9.prism.bin
04-05 06:24:03.745 I/rime.trime(17538): found double array image of size 6656.
04-05 06:24:03.877 I/rime.trime(17538): stabledb 'custom_phrase_t9' does not exist.
04-05 06:24:03.878 I/rime.trime(17538): initializing opencc: /storage/emulated/0/rime/opencc/emoji.json
04-05 06:24:03.914 I/rime.trime(17538): initializing opencc: /storage/emulated/0/Android/data/com.osfans.trime/files/shared/opencc/s2t.json
04-05 06:24:03.964 I/rime.trime(17538): ConcreteEngine::InitializeOptions
04-05 06:24:03.964 I/rime.trime(17538): found switch option: ascii_mode, reset: -1
04-05 06:24:03.964 I/rime.trime(17538): found switch option: ascii_punct, reset: -1
04-05 06:24:03.964 I/rime.trime(17538): found switch option: traditionalization, reset: -1
04-05 06:24:03.964 I/rime.trime(17538): found switch option: emoji, reset: 1
04-05 06:24:03.964 I/rime.trime(17538): updated option: emoji
04-05 06:24:03.964 I/rime.trime(17538): found switch option: full_shape, reset: -1
04-05 06:24:03.964 I/rime.trime(17538): found switch option: search_single_char, reset: -1
04-05 06:24:03.965 I/[rime-main](17538): Loading config file /storage/emulated/0/rime/build/t9.schema.yaml
04-05 06:24:03.965 E/SpannableStringBuilder(17538): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
04-05 06:24:03.965 E/SpannableStringBuilder(17538): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
04-05 06:24:05.130 D/VRI[PrefLauncherAlias](17538): onFocusEvent false
04-05 06:24:05.980 D/ScrollOptim [SceneManager](17538): updateCurrentActivity: mCurrentActivityName=null, isOptEnable=true, isAnimAheadEnable=true, isFrameInsertEnable=true, InsertNum=1, isEnabledForScrollChanged=false
04-05 06:24:05.990 D/VRI[PrefLauncherAlias](17538): dispatchAppVisibility visible:false
04-05 06:24:05.990 D/VRI[PrefLauncherAlias](17538): visibilityChanged oldVisibility=true newVisibility=false
04-05 06:24:05.994 D/HWUI    (17538): RenderProxy::destroy: this=0xb4000079ba964000, mContext=0xb4000079ba91a000
04-05 06:24:05.994 D/HWUI    (17538): SkiaOpenGLPipeline::setSurface: this=0xb4000079ba8b09c0, surface=NULL
04-05 06:24:05.995 D/BufferQueueProducer(17538): [VRI[PrefLauncherAlias]#0(BLAST Consumer)0](id:448200000000,api:1,p:17538,c:17538) disconnect: api 1
04-05 06:24:05.995 D/ViewRootImplExtImpl(17538): wrapConfigInfoIntoFlags rotation=0, smallestScreenWidthDp=360, relayoutAsync=false, flags=0, newFlags=23592960, residentWS=false, title=com.osfans.trime/com.osfans.trime.PrefLauncherAlias
--------- beginning of system
04-05 06:24:06.001 D/VRI[PrefLauncherAlias](17538): relayoutWindow result, sizeChanged:false, surfaceControlChanged:true, transformHintChanged:false, mSurfaceSize:Point(1080, 2414), mLastSurfaceSize:Point(1080, 2414), mWidth:1080, mHeight:2414, requestedWidth:1080, requestedHeight:2414, transformHint:0, installOrientation:0, displayRotation:0, isSurfaceValid:false, attr.flag:-2122252032, tmpFrames:ClientWindowFrames{frame=[0,0][1080,2414] display=[0,0][1080,2414] parentFrame=[0,0][0,0]}, relayoutAsync:false, mSyncSeqId:0
04-05 06:24:06.001 D/BLASTBufferQueue(17538): [VRI[PrefLauncherAlias]#0](f:0,a:1) destructor()
04-05 06:24:06.001 D/BufferQueueConsumer(17538): [VRI[PrefLauncherAlias]#0(BLAST Consumer)0](id:448200000000,api:0,p:-1,c:17538) disconnect
04-05 06:24:06.003 D/VRI[PrefLauncherAlias](17538): setWindowStopped stopped:true
04-05 06:24:06.010 D/ViewRootImplExtImpl(17538): wrapConfigInfoIntoFlags rotation=0, smallestScreenWidthDp=360, relayoutAsync=false, flags=0, newFlags=23592960, residentWS=false, title=com.osfans.trime/com.osfans.trime.PrefLauncherAlias
04-05 06:24:06.011 I/om.osfans.trime(17538): _Desupression native alloc gc_urgency: 80006 native bytes: 36764849
04-05 06:24:06.014 D/VRI[PrefLauncherAlias](17538): relayoutWindow result, sizeChanged:false, surfaceControlChanged:true, transformHintChanged:false, mSurfaceSize:Point(1080, 2414), mLastSurfaceSize:Point(1080, 2414), mWidth:1080, mHeight:2414, requestedWidth:1080, requestedHeight:2414, transformHint:0, installOrientation:0, displayRotation:0, isSurfaceValid:false, attr.flag:-2122252032, tmpFrames:ClientWindowFrames{frame=[0,0][1080,2414] display=[0,0][1080,2414] parentFrame=[0,0][0,0]}, relayoutAsync:false, mSyncSeqId:0
04-05 06:24:06.048 W/om.osfans.trime(17538): Cleared Reference was only reachable from finalizer (only reported once)
04-05 06:24:06.135 D/OplusScrollToTopManager(17538): com.osfans.trime/com.osfans.trime.PrefLauncherAlias,This com.android.internal.policy.DecorView{fc4b3da I.E...... R.....ID 0,0-1080,2414 alpha=1.0 viewInfo = }[PrefLauncherAlias] change focus to false
04-05 06:24:07.374 D/AndroidRuntime(17538): Shutting down VM
--------- beginning of crash
04-05 06:24:07.374 E/AndroidRuntime(17538): FATAL EXCEPTION: main
04-05 06:24:07.374 E/AndroidRuntime(17538): Process: com.osfans.trime, PID: 17538
04-05 06:24:07.374 E/AndroidRuntime(17538): androidx.startup.StartupException: lateinit property _activeTheme has not been initialized
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at com.osfans.trime.data.theme.ThemeManager.getActiveTheme(SourceFile:8)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at com.osfans.trime.ime.core.TrimeInputMethodService.onCreateInputView(SourceFile:11)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2449)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:3307)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:3224)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:1051)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:1012)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.os.Handler.dispatchMessage(Handler.java:107)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.os.Looper.loopOnce(Looper.java:282)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.os.Looper.loop(Looper.java:387)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at android.app.ActivityThread.main(ActivityThread.java:9500)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at java.lang.reflect.Method.invoke(Native Method)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
04-05 06:24:07.374 E/AndroidRuntime(17538): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)
04-05 06:24:07.401 I/om.osfans.trime(17538): System.exit called, status: 10
04-05 06:24:07.401 I/AndroidRuntime(17538): VM exiting with result code 10, cleanup skipped.
```

```
--------- Device Info
OS Name: PJF110_15.0.0.701(CN01)
OS Version: 6.1.75-android14-11-o-g74387f2552b6 (U.1cf6eb4_1-8c69)
OS API Level: 35
Device: OP5CFBL1
Model (product): PJF110 (PJF110)
Manufacturer: OnePlus
Tags: release-keys
Screen Size: 1080 x 2245
Screen Density: 3.0
Screen orientation: Portrait
--------- Build Info
Package Name: com.osfans.trime
Builder: (Unknown)
Version Code: 20250301
Version Name: v3.3.3-0-g9e3689b3-release
Build Time: 2025-03-03T02:08:54Z
Build Git Hash: 9e3689b34c8fcf27bfa3f1e11cb3d79a33ec11c3
--------- Crash stacktrace
java.lang.RuntimeException: Unable to stop service com.osfans.trime.ime.core.TrimeInputMethodService@f93d2ea: java.lang.IllegalArgumentException: Receiver not registered: com.osfans.trime.receiver.RimeIntentReceiver@5fca079
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5583)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2727)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:282)
	at android.os.Looper.loop(Looper.java:387)
	at android.app.ActivityThread.main(ActivityThread.java:9500)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)
Caused by: java.lang.IllegalArgumentException: Receiver not registered: com.osfans.trime.receiver.RimeIntentReceiver@5fca079
	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1777)
	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1922)
	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:832)
	at com.osfans.trime.ime.core.TrimeInputMethodService.onDestroy(SourceFile:96)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5563)
	... 9 more

--------- beginning of system
04-05 06:26:26.706 D/ResourcesManagerExtImpl(20846): applyConfigurationToAppResourcesLocked app.getDisplayId() return callback.displayId:-1
04-05 06:26:26.724 D/ResourcesManagerExtImpl(20846): applyConfigurationToAppResourcesLocked app.getDisplayId() return callback.displayId:0
04-05 06:26:26.740 E/OplusCustomizeRestrictionManager(20846): sInstance is null, start a new sInstance
04-05 06:26:26.746 D/OplusInputMethodUtil(20846): init sDebug to false, init sDebugIme to false, init sAlwaysOn to false
04-05 06:26:26.746 D/OplusInputMethodUtil(20846): updateDebugToClass InputMethodManager.DEBUG = false
04-05 06:26:26.746 D/OplusInputMethodUtil(20846): updateDebugToClass ImeFocusController.DEBUG = false
04-05 06:26:26.746 D/OplusInputMethodUtil(20846): updateDebugToClass InsetsController.DEBUG = false
04-05 06:26:26.746 D/OplusInputMethodUtil(20846): updateDebugToClass BaseInputConnection.DEBUG = false
04-05 06:26:26.770 D/ActivityThread(20846): ComponentInfo{com.osfans.trime/com.osfans.trime.ui.main.LogActivity} checkFinished=false 2
04-05 06:26:26.770 D/ResourcesManagerExtImpl(20846): applyConfigurationToAppResourcesLocked app.getDisplayId() return callback.displayId:0
04-05 06:26:26.777 D/ResourcesManagerExtImpl(20846): applyConfigurationToAppResourcesLocked app.getDisplayId() return callback.displayId:0
04-05 06:26:26.778 D/OplusInputMethodUtil(20846): updateDebugToClass InputMethodService.DEBUG = false
04-05 06:26:26.858 D/VRI[LogActivity](20846): relayoutWindow result, sizeChanged:true, surfaceControlChanged:true, transformHintChanged:false, mSurfaceSize:Point(1080, 2414), mLastSurfaceSize:Point(0, 0), mWidth:-1, mHeight:-1, requestedWidth:1080, requestedHeight:2414, transformHint:0, installOrientation:0, displayRotation:0, isSurfaceValid:true, attr.flag:-2122252032, tmpFrames:ClientWindowFrames{frame=[0,0][1080,2414] display=[0,0][1080,2414] parentFrame=[0,0][0,0]}, relayoutAsync:false, mSyncSeqId:0
04-05 06:26:26.859 W/VRI[LogActivity](20846): updateBlastSurfaceIfNeeded, surfaceSize:Point(1080, 2414), lastSurfaceSize:Point(0, 0), format:-1, blastBufferQueue:null
04-05 06:26:27.284 E/ActivityThread(20846): Failed to find provider info for com.google.android.gms.fonts
04-05 06:26:28.154 D/OplusInputMethodManagerInternal(20846): get inputMethodManager extension: com.android.internal.view.IInputMethodManager$Stub$Proxy@9109ae0
04-05 06:26:29.196 D/VRI[LogActivity](20846): relayoutWindow result, sizeChanged:false, surfaceControlChanged:true, transformHintChanged:false, mSurfaceSize:Point(1080, 2414), mLastSurfaceSize:Point(1080, 2414), mWidth:1080, mHeight:2414, requestedWidth:1080, requestedHeight:2414, transformHint:0, installOrientation:0, displayRotation:0, isSurfaceValid:false, attr.flag:-2122252032, tmpFrames:ClientWindowFrames{frame=[0,0][1080,2414] display=[0,0][1080,2414] parentFrame=[0,0][0,0]}, relayoutAsync:false, mSyncSeqId:0
--------- beginning of main
04-05 06:26:29.710 W/om.osfans.trime(20846): Cleared Reference was only reachable from finalizer (only reported once)
04-05 06:26:29.725 D/HWUI    (20846): SkiaOpenGLPipeline::setSurface: this=0xb400007b5ff1e480, surface=NULL
04-05 06:26:29.725 D/HWUI    (20846): SkiaOpenGLPipeline::setSurface: this=0xb400007b5ff1e340, surface=NULL
04-05 06:26:37.120 D/Vibrator(20846): SystemVibrator Created
04-05 06:26:37.146 D/OplusScrollToTopManager(20846): InputMethod,window dying
04-05 06:26:37.147 D/OplusScrollToTopManager(20846): InputMethod,unregisterSystemUIBroadcastReceiver 
04-05 06:26:37.147 D/OplusScrollToTopManager(20846): InputMethod, unregisterSystemUIBroadcastReceiver failed java.lang.IllegalArgumentException: Receiver not registered: android.view.OplusScrollToTopManager$2@506b072
04-05 06:26:37.147 W/WindowOnBackDispatcher(20846): sendCancelIfRunning: isInProgress=false callback=android.view.ViewRootImpl$$ExternalSyntheticLambda11@40bd62e
04-05 06:26:37.149 D/HWUI    (20846): RenderProxy::destroy: this=0xb4000079bbd9b500, mContext=0xb4000079bbc78000
04-05 06:26:37.149 D/HWUI    (20846): SkiaOpenGLPipeline::setSurface: this=0xb4000079bbdb8f80, surface=NULL
04-05 06:26:37.160 D/AndroidRuntime(20846): Shutting down VM
--------- beginning of crash
04-05 06:26:37.160 E/AndroidRuntime(20846): FATAL EXCEPTION: main
04-05 06:26:37.160 E/AndroidRuntime(20846): Process: com.osfans.trime, PID: 20846
04-05 06:26:37.160 E/AndroidRuntime(20846): java.lang.RuntimeException: Unable to stop service com.osfans.trime.ime.core.TrimeInputMethodService@f93d2ea: java.lang.IllegalArgumentException: Receiver not registered: com.osfans.trime.receiver.RimeIntentReceiver@5fca079
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.ActivityThread.handleStopService(ActivityThread.java:5583)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2727)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.os.Handler.dispatchMessage(Handler.java:107)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.os.Looper.loopOnce(Looper.java:282)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.os.Looper.loop(Looper.java:387)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.ActivityThread.main(ActivityThread.java:9500)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at java.lang.reflect.Method.invoke(Native Method)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)
04-05 06:26:37.160 E/AndroidRuntime(20846): Caused by: java.lang.IllegalArgumentException: Receiver not registered: com.osfans.trime.receiver.RimeIntentReceiver@5fca079
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1777)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1922)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:832)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at com.osfans.trime.ime.core.TrimeInputMethodService.onDestroy(SourceFile:96)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	at android.app.ActivityThread.handleStopService(ActivityThread.java:5563)
04-05 06:26:37.160 E/AndroidRuntime(20846): 	... 9 more
04-05 06:26:37.185 I/om.osfans.trime(20846): System.exit called, status: 10
04-05 06:26:37.185 I/AndroidRuntime(20846): VM exiting with result code 10, cleanup skipped.
```

--- a/app/src/main/java/com/osfans/trime/data/base/DataManager.kt
+++ b/app/src/main/java/com/osfans/trime/data/base/DataManager.kt
@@ -100,10 +100,17 @@ object DataManager {
 
             ResourceUtils.copyFile(DATA_CHECKSUMS_NAME, dataDir.resolve(DATA_CHECKSUMS_NAME).absolutePath)
 
-            // 创建空的 default.custom.yaml 以覆盖 default.yaml 默认的方案列表
+            // 创建 default.custom.yaml 并用空方案列表覆盖 default.yaml 默认的方案列表
+            // TODO: 内置明月拼音等方案，提供最小开箱即用环境
             runCatching {
-                if (File(userDataDir, DEFAULT_CUSTOM_FILE_NAME).createNewFile()) {
-                    Timber.d("Created empty default.custom.yaml")
+                val defaultCustom = File(userDataDir, DEFAULT_CUSTOM_FILE_NAME)
+                if (defaultCustom.createNewFile()) {
+                    defaultCustom.writeText(
+                        """
+                        patch:
+                            schema_list: []
+                        """.trimIndent(),
+                    )
                 }
             }.getOrElse { Timber.e(it, "Failed to create default.custom.yaml") }
 

--- a/app/src/main/java/com/osfans/trime/data/db/ClipboardHelper.kt
+++ b/app/src/main/java/com/osfans/trime/data/db/ClipboardHelper.kt
@@ -6,7 +6,9 @@ package com.osfans.trime.data.db
 
 import android.content.ClipboardManager
 import android.content.Context
+import android.os.Build
 import androidx.room.Room
+import androidx.room.withTransaction
 import com.osfans.trime.data.prefs.AppPrefs
 import com.osfans.trime.util.WeakHashSet
 import com.osfans.trime.util.matchesAny
@@ -49,21 +51,25 @@ object ClipboardHelper :
         onUpdateListeners.remove(listener)
     }
 
-    private val limit get() = AppPrefs.defaultInstance().clipboard.clipboardLimit
-    private val compare get() =
-        AppPrefs
-            .defaultInstance()
-            .clipboard.clipboardCompareRules
+    private val clipPref = AppPrefs.defaultInstance().clipboard
+
+    private val limit by clipPref.clipboardLimit
+
+    private val compareRules: Set<Regex> by lazy {
+        val rules by clipPref.clipboardCompareRules
+        rules
             .split('\n')
             .map { Regex(it.trim()) }
-            .toHashSet()
-    private val output get() =
-        AppPrefs
-            .defaultInstance()
-            .clipboard.clipboardOutputRules
+            .toSet()
+    }
+
+    private val outputRules: Set<Regex> by lazy {
+        val rules by clipPref.clipboardOutputRules
+        rules
             .split('\n')
             .map { Regex(it) }
-            .toHashSet()
+            .toSet()
+    }
 
     var lastBean: DatabaseBean? = null
 
@@ -117,41 +123,63 @@ object ClipboardHelper :
         updateItemCount()
     }
 
+    private var lastClipTimestamp = -1L
+    private var lastClipHash = 0
+
     /**
      * 此方法设置监听剪贴板变化，如有新的剪贴内容，就启动选定的剪贴板管理器
      *
-     * - [compare] 比较规则。每次通知剪贴板管理器，都会保存 ClipBoardCompare 处理过的 string。
+     * - [compareRules] 比较规则。每次通知剪贴板管理器，都会保存 ClipBoardCompare 处理过的 string。
      * 如果两次处理过的内容不变，则不通知。
      *
-     * - [output] 输出规则。如果剪贴板内容与规则匹配，则不通知剪贴板管理器。
+     * - [outputRules] 输出规则。如果剪贴板内容与规则匹配，则不通知剪贴板管理器。
      */
     override fun onPrimaryClipChanged() {
         if (!(limit != 0 && this::clbDao.isInitialized)) {
             return
         }
-        clipboardManager
-            .primaryClip
-            ?.let { DatabaseBean.fromClipData(it) }
-            ?.takeIf {
-                it.text!!.isNotBlank() &&
-                    !it.text.matchesAny(output)
-            }?.let { b ->
-                if (b.text!!.removeRegexSet(compare).isEmpty()) return
-                Timber.d("Accept clipboard $b")
-                launch {
-                    mutex.withLock {
-                        clbDao.find(b.text)?.let {
-                            clbDao.updateTime(it.id, b.time)
-                            updateLastBean(it.copy(time = b.time))
-                            return@launch
-                        }
-                        val rowId = clbDao.insert(b)
-                        removeOutdated()
-                        updateItemCount()
-                        updateLastBean(clbDao.get(rowId) ?: b)
+        val clip = clipboardManager.primaryClip ?: return
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val timestamp = clip.description.timestamp
+            if (timestamp == lastClipTimestamp) return
+            lastClipTimestamp = timestamp
+        } else {
+            val timestamp = System.currentTimeMillis()
+            val hash = clip.hashCode()
+            if (timestamp - lastClipTimestamp < 100L && hash == lastClipHash) return
+            lastClipTimestamp = timestamp
+            lastClipHash = hash
+        }
+        launch {
+            mutex.withLock {
+                val bean = DatabaseBean.fromClipData(clip) ?: return@withLock
+                if (bean.text.isNullOrBlank()) return@withLock
+                if (bean.text.matchesAny(outputRules) ||
+                    bean.text.removeRegexSet(compareRules).isEmpty()
+                ) {
+                    return@withLock
+                }
+                try {
+                    clbDao.find(bean.text)?.let {
+                        updateLastBean(it.copy(time = bean.time))
+                        clbDao.updateTime(it.id, bean.time)
+                        return@withLock
                     }
+                    val insertedBean =
+                        clbDb.withTransaction {
+                            val rowId = clbDao.insert(bean)
+                            removeOutdated()
+                            updateItemCount()
+                            clbDao.get(rowId) ?: bean
+                        }
+                    updateLastBean(insertedBean)
+                    updateItemCount()
+                } catch (exception: Exception) {
+                    Timber.w("Failed to update clipboard database: $exception")
+                    updateLastBean(bean)
                 }
             }
+        }
     }
 
     private suspend fun removeOutdated() {

--- a/app/src/main/java/com/osfans/trime/data/prefs/AppPrefs.kt
+++ b/app/src/main/java/com/osfans/trime/data/prefs/AppPrefs.kt
@@ -244,10 +244,10 @@ class AppPrefs(
             const val CLIPBOARD_LIMIT = "clipboard_clipboard_limit"
         }
 
-        var clipboardCompareRules by string(CLIPBOARD_COMPARE_RULES, "")
-        var clipboardOutputRules by string(CLIPBOARD_OUTPUT_RULES, "")
+        val clipboardCompareRules = string(CLIPBOARD_COMPARE_RULES, "")
+        val clipboardOutputRules = string(CLIPBOARD_OUTPUT_RULES, "")
         var draftOutputRules by string(DRAFT_OUTPUT_RULES, "")
-        var clipboardLimit by int(CLIPBOARD_LIMIT, 10)
+        val clipboardLimit = int(CLIPBOARD_LIMIT, 10)
         var draftLimit by int(DRAFT_LIMIT, 10)
         var draftExcludeApp by string(DRAFT_EXCLUDE_APP, "")
     }

--- a/app/src/main/java/com/osfans/trime/data/schema/Schema.kt
+++ b/app/src/main/java/com/osfans/trime/data/schema/Schema.kt
@@ -31,6 +31,7 @@ data class Schema(
 
     companion object {
         fun decodeBySchemaId(schemaId: String): Schema {
+            if (schemaId == ".default") return Schema()
             val file = File(DataManager.resolveDeployedResourcePath("$schemaId.schema"))
             val root = SchemaManager.yaml.parseToYamlNode(file.readText()).yamlMap
             return Schema(

--- a/app/src/main/java/com/osfans/trime/data/schema/SchemaManager.kt
+++ b/app/src/main/java/com/osfans/trime/data/schema/SchemaManager.kt
@@ -21,13 +21,16 @@ object SchemaManager {
                 ),
         )
 
-    @JvmStatic
     fun init(schemaId: String) {
-        currentSchema = Schema.decodeBySchemaId(schemaId)
-        visibleSwitches =
-            currentSchema.switches
-                .filter { it.states.isNotEmpty() } // 剔除没有 states 条目项的值，它们不作为开关使用
-        updateSwitchOptions()
+        runCatching {
+            currentSchema = Schema.decodeBySchemaId(schemaId)
+            visibleSwitches =
+                currentSchema.switches
+                    .filter { it.states.isNotEmpty() } // 剔除没有 states 条目项的值，它们不作为开关使用
+            updateSwitchOptions()
+        }.getOrElse {
+            Timber.w(it, "Failed to decode schema file of id '$schemaId'")
+        }
     }
 
     val activeSchema: Schema

--- a/app/src/main/java/com/osfans/trime/data/theme/ColorManager.kt
+++ b/app/src/main/java/com/osfans/trime/data/theme/ColorManager.kt
@@ -168,10 +168,10 @@ object ColorManager {
         val color =
             try {
                 resolveValue(key) { value ->
-                    parseColor(value)
+                    ColorUtils.parseColor(value)
                 }
             } catch (_: IllegalArgumentException) {
-                parseColor(key)
+                ColorUtils.parseColor(key)
             }
         if (putCache) {
             synchronized(colorCache) { colorCache.put(key, color) }
@@ -222,14 +222,6 @@ object ColorManager {
             }
         }
     }
-
-    private fun parseColor(value: String): Int =
-        try {
-            ColorUtils.parseColor(value)
-        } catch (e: Exception) {
-            Timber.e(e, "Cannot parse color $value, fallback to TRANSPARENT")
-            Color.TRANSPARENT
-        }
 
     private fun parseDrawable(value: String): Drawable? {
         if (value.isEmpty()) return null

--- a/app/src/main/java/com/osfans/trime/data/theme/mapper/TextKeyboardMapper.kt
+++ b/app/src/main/java/com/osfans/trime/data/theme/mapper/TextKeyboardMapper.kt
@@ -42,12 +42,12 @@ class TextKeyboardMapper(
                         keyHintOffsetY = it.getFloat("key_hint_offset_y"),
                         keyPressOffsetX = it.getInt("key_press_offset_x"),
                         keyPressOffsetY = it.getInt("key_press_offset_y"),
-                        keyTextColor = it.getString("key_text_color", "key_text_color"),
-                        keyBackColor = it.getString("key_back_color", "key_back_color"),
-                        keySymbolColor = it.getString("key_symbol_color", "key_symbol_color"),
-                        hlKeyTextColor = it.getString("hilited_key_text_color", "hilited_key_text_color"),
-                        hlKeyBackColor = it.getString("hilited_key_back_color", "hilited_key_back_color"),
-                        hlKeySymbolColor = it.getString("hilited_key_symbol_color", "hilited_key_symbol_color"),
+                        keyTextColor = it.getString("key_text_color"),
+                        keyBackColor = it.getString("key_back_color"),
+                        keySymbolColor = it.getString("key_symbol_color"),
+                        hlKeyTextColor = it.getString("hilited_key_text_color"),
+                        hlKeyBackColor = it.getString("hilited_key_back_color"),
+                        hlKeySymbolColor = it.getString("hilited_key_symbol_color"),
                         behaviors =
                             buildMap {
                                 KeyBehavior.entries.forEach { entry ->

--- a/app/src/main/java/com/osfans/trime/ime/keyboard/Key.kt
+++ b/app/src/main/java/com/osfans/trime/ime/keyboard/Key.kt
@@ -72,16 +72,16 @@ class Key(
         fallback: String,
     ): Int =
         selfConfig?.let {
-            ColorManager
-                .getColor(src(selfConfig))
-                .takeIf { it != Color.TRANSPARENT }
+            runCatching { ColorManager.getColor(src(it)) }.getOrNull()
         } ?: ColorManager.getColor(fallback)
 
     private fun getDrawable(
         src: TextKeyboard.TextKey.() -> String,
         fallback: String,
-    ) = selfConfig?.let { ColorManager.getDrawable(src(it)) }
-        ?: ColorManager.getDrawable(fallback)
+    ) = selfConfig?.let {
+        if (src(it).isEmpty()) null
+        ColorManager.getDrawable(src(it))
+    } ?: ColorManager.getDrawable(fallback)
 
     private val keyBackground by lazy { getDrawable({ keyBackColor }, "key_back_color") }
     private val offKeyBackground by lazy { ColorManager.getDrawable("off_key_back_color") }

--- a/app/src/main/java/com/osfans/trime/ui/main/PrefMainActivity.kt
+++ b/app/src/main/java/com/osfans/trime/ui/main/PrefMainActivity.kt
@@ -7,13 +7,14 @@ package com.osfans.trime.ui.main
 import android.content.Intent
 import android.os.Bundle
 import android.view.Menu
-import android.view.MenuItem
 import android.view.ViewGroup
 import androidx.activity.enableEdgeToEdge
 import androidx.activity.viewModels
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.app.AppCompatDelegate
+import androidx.appcompat.graphics.drawable.DrawerArrowDrawable
+import androidx.core.content.ContextCompat
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowCompat
 import androidx.core.view.WindowInsetsCompat
@@ -67,7 +68,16 @@ class PrefMainActivity : AppCompatActivity() {
             .isAppearanceLightStatusBars = false
 
         setContentView(binding.root)
-        setSupportActionBar(binding.prefToolbar.toolbar)
+        // always show toolbar back arrow icon
+        binding.prefToolbar.toolbar.navigationIcon =
+            DrawerArrowDrawable(this).apply {
+                progress = 1f
+                color = ContextCompat.getColor(this@PrefMainActivity, R.color.colorOnPrimary)
+            }
+        // show menu icon and other action icons on toolbar
+        // don't use `setSupportActionBar(binding.toolbar)` here,
+        // because navController would change toolbar title, we need to control it by ourselves
+        setupToolbarMenu(binding.prefToolbar.toolbar.menu)
         navController = binding.navHostFragment.getFragment<NavHostFragment>().navController
         binding.prefToolbar.toolbar.setNavigationOnClickListener {
             // prevent navigate up when child fragment has enabled `OnBackPressedCallback`
@@ -99,55 +109,50 @@ class PrefMainActivity : AppCompatActivity() {
         checkNotificationPermission()
     }
 
-    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+    private fun setupToolbarMenu(menu: Menu) {
         menuInflater.inflate(R.menu.main_menu, menu)
-        menu?.forEach { item ->
+        menu.forEach { item ->
+            // show menu item on demand
+            item.isVisible = false
             when (item.itemId) {
                 R.id.deploy, R.id.about -> {
                     viewModel.topOptionsMenu.observe(this) {
                         item.isVisible = it
+                    }
+                    if (item.itemId == R.id.deploy) {
+                        item.setOnMenuItemClickListener {
+                            lifecycleScope.launch { RimeDaemon.restartRime(true) }
+                            true
+                        }
+                    } else if (item.itemId == R.id.about) {
+                        item.setOnMenuItemClickListener {
+                            navController.navigate(R.id.action_prefFragment_to_aboutFragment)
+                            true
+                        }
                     }
                 }
                 R.id.edit -> {
                     viewModel.toolbarEditButtonVisible.observe(this) {
                         item.isVisible = it
                     }
+                    item.setOnMenuItemClickListener {
+                        viewModel.toolbarEditButtonOnClickListener.value?.invoke()
+                        true
+                    }
                 }
                 R.id.delete -> {
                     viewModel.toolbarDeleteButtonOnClickListener.observe(this) {
                         item.isVisible = it != null
                     }
+                    item.setOnMenuItemClickListener {
+                        viewModel.toolbarDeleteButtonOnClickListener.value?.invoke()
+                        true
+                    }
                 }
                 else -> {}
             }
         }
-        // show menu item on demand
-        menu?.forEach { it.isVisible = false }
-        return true
     }
-
-    override fun onOptionsItemSelected(item: MenuItem): Boolean =
-        when (item.itemId) {
-            R.id.deploy -> {
-                lifecycleScope.launch {
-                    RimeDaemon.restartRime(true)
-                }
-                true
-            }
-            R.id.about -> {
-                navController.navigate(R.id.action_prefFragment_to_aboutFragment)
-                true
-            }
-            R.id.edit -> {
-                viewModel.toolbarEditButtonOnClickListener.value?.invoke()
-                true
-            }
-            R.id.delete -> {
-                viewModel.toolbarDeleteButtonOnClickListener.value?.invoke()
-                true
-            }
-            else -> super.onOptionsItemSelected(item)
-        }
 
     override fun onPause() {
         super.onPause()

--- a/app/src/main/java/com/osfans/trime/util/ColorUtils.kt
+++ b/app/src/main/java/com/osfans/trime/util/ColorUtils.kt
@@ -4,11 +4,11 @@
 
 package com.osfans.trime.util
 
-import android.graphics.Color
 import androidx.annotation.ColorInt
 import androidx.core.graphics.blue
 import androidx.core.graphics.green
 import androidx.core.graphics.red
+import androidx.core.graphics.toColorInt
 
 object ColorUtils {
     @ColorInt
@@ -25,7 +25,7 @@ object ColorUtils {
             } else {
                 colorString // red, green, blue ...
             }
-        return Color.parseColor(normalized)
+        return normalized.toColorInt()
     }
 
     /**

--- a/app/src/main/res/drawable/ic_baseline_delete_24.xml
+++ b/app/src/main/res/drawable/ic_baseline_delete_24.xml
@@ -4,8 +4,11 @@ SPDX-FileCopyrightText: 2015 - 2024 Rime community
 SPDX-License-Identifier: GPL-3.0-or-later
 -->
 
-<vector android:alpha="0.8" android:height="24dp" android:tint="#000000"
-    android:viewportHeight="24" android:viewportWidth="24"
-    android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+<vector android:height="24dp"
+    android:tint="?attr/colorControlNormal"
+    android:viewportHeight="24"
+    android:viewportWidth="24"
+    android:width="24dp"
+    xmlns:android="http://schemas.android.com/apk/res/android">
     <path android:fillColor="@android:color/white" android:pathData="M6,19c0,1.1 0.9,2 2,2h8c1.1,0 2,-0.9 2,-2V7H6v12zM19,4h-3.5l-1,-1h-5l-1,1H5v2h14V4z"/>
 </vector>

--- a/app/src/main/res/drawable/ic_baseline_edit_24.xml
+++ b/app/src/main/res/drawable/ic_baseline_edit_24.xml
@@ -4,8 +4,11 @@ SPDX-FileCopyrightText: 2015 - 2024 Rime community
 SPDX-License-Identifier: GPL-3.0-or-later
 -->
 
-<vector android:height="24dp" android:tint="#000000"
-    android:viewportHeight="24" android:viewportWidth="24"
-    android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+<vector android:height="24dp"
+    android:tint="?attr/colorControlNormal"
+    android:viewportHeight="24"
+    android:viewportWidth="24"
+    android:width="24dp"
+    xmlns:android="http://schemas.android.com/apk/res/android">
     <path android:fillColor="@android:color/white" android:pathData="M3,17.25V21h3.75L17.81,9.94l-3.75,-3.75L3,17.25zM20.71,7.04c0.39,-0.39 0.39,-1.02 0,-1.41l-2.34,-2.34c-0.39,-0.39 -1.02,-0.39 -1.41,0l-1.83,1.83 3.75,3.75 1.83,-1.83z"/>
 </vector>

--- a/app/src/main/res/drawable/ic_baseline_sync_24.xml
+++ b/app/src/main/res/drawable/ic_baseline_sync_24.xml
@@ -5,7 +5,10 @@ SPDX-License-Identifier: GPL-3.0-or-later
 -->
 
 <vector android:autoMirrored="true" android:height="24dp"
-    android:tint="#000000" android:viewportHeight="24"
-    android:viewportWidth="24" android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+    android:tint="?attr/colorControlNormal"
+    android:viewportHeight="24"
+    android:viewportWidth="24"
+    android:width="24dp"
+    xmlns:android="http://schemas.android.com/apk/res/android">
     <path android:fillColor="@android:color/white" android:pathData="M12,4L12,1L8,5l4,4L12,6c3.31,0 6,2.69 6,6 0,1.01 -0.25,1.97 -0.7,2.8l1.46,1.46C19.54,15.03 20,13.57 20,12c0,-4.42 -3.58,-8 -8,-8zM12,18c-3.31,0 -6,-2.69 -6,-6 0,-1.01 0.25,-1.97 0.7,-2.8L5.24,7.74C4.46,8.97 4,10.43 4,12c0,4.42 3.58,8 8,8v3l4,-4 -4,-4v3z"/>
 </vector>

--- a/app/src/main/res/menu/main_menu.xml
+++ b/app/src/main/res/menu/main_menu.xml
@@ -11,18 +11,15 @@ SPDX-License-Identifier: GPL-3.0-or-later
     <item android:id="@+id/deploy"
         android:title="@string/pref_deploy"
         android:icon="@drawable/ic_baseline_sync_24"
-        app:iconTint="?attr/colorControlNormal"
         app:showAsAction="ifRoom" />
     <item android:id="@+id/about"
         android:title="@string/settings__about_title" />
     <item android:id="@+id/edit"
         android:icon="@drawable/ic_baseline_edit_24"
-        app:iconTint="?attr/colorControlNormal"
         android:title="@string/edit"
         app:showAsAction="ifRoom" />
     <item android:id="@+id/delete"
         android:icon="@drawable/ic_baseline_delete_24"
-        app:iconTint="?attr/colorControlNormal"
         android:title="@string/delete"
         app:showAsAction="ifRoom" />
 </menu>

